src/ai/susi/mind/SusiCognition.java

```diff
@@ -75,14 +78,89 @@ public SusiCognition(
         // compute the mind's reaction: here we compute with a hierarchy of minds. The dispute is taken from the relevant mind level that was able to compute the dispute
         List<SusiThought> dispute = SusiMind.reactMinds(query, language, maxcount, client, observation, minds);
         long answer_date = System.currentTimeMillis();
+
+        if (!countryCode.equals("") && !countryName.equals("")) {
+            List<String> skills = dispute.get(0).getSkills();
+            for (String skill : skills) {
+                try {
+                    updateCountryWiseUsageData(skill, countryCode, countryName);
+                }
+                catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
         
         // store answer and actions into json
         this.json.put("answers", new JSONArray(dispute));
         this.json.put("answer_date", DateParser.utcFormatter.print(answer_date));
         this.json.put("answer_time", answer_date - query_date);
         this.json.put("language", language.name());
     }
-    
+
+    private void updateCountryWiseUsageData(String skillPath, String countryCode, String countryName) {
+        String skillInfo[] = skillPath.split("/");
+        String model_name = skillInfo[3];
+        String group_name = skillInfo[4];
+        String language_name = skillInfo[5];
+        String skill_name = skillInfo[6].split("\\.")[0];
+        JsonTray skillUsage = DAO.countryWiseSkillUsage;
+        JSONObject modelName = new JSONObject();
+        JSONObject groupName = new JSONObject();
+        JSONObject languageName = new JSONObject();
+        if (skillUsage.has(model_name)) {
+            modelName = skillUsage.getJSONObject(model_name);
+            if (modelName.has(group_name)) {
+                groupName = modelName.getJSONObject(group_name);
+                if (groupName.has(language_name)) {
+                    languageName = groupName.getJSONObject(language_name);
+                    if (languageName.has(skill_name)) {
+                        JSONArray countryWiseUsageData = languageName.getJSONArray(skill_name);
+                        Boolean countryExists = false;
+                        for (int i = 0; i<countryWiseUsageData.length(); i++) {
+                            JSONObject countryUsage = countryWiseUsageData.getJSONObject(i);
+                            if (countryUsage.get("country_code").equals(countryCode)){
+                                countryUsage.put("count", countryUsage.getInt("count")+1);
+                                countryWiseUsageData.put(i,countryUsage);
+                                countryExists = true;
+                                break;
+                            }
+                        }
+
+                        if (!countryExists) {
```

Comment:
It is getting redundant here. You can directly use the block that is present outside the `if` sections. I would suggest you to take a look at `FiveStarRateSkillService.java` for reference.